### PR TITLE
Update Sntp_ReceiveTimeResponse to rotate server on response timeout

### PR DESCRIPTION
**Pull request by aggarw13**
### Major changes

* Update behavior of `Sntp_ReceiveTimeResponse` API to additionally rotate server when there is a timeout in receiving server.
response. 
* Update the server rotation logic to _wrap around_ the server list once all servers have been used in the rotation occurrences. This is done to avoid the SNTP client in a device from becoming dysfunctional if such a scenario is hit.

### Other changes
* Update `calculateElapsedTimeMs` to address integer overflow issue flagged by CBMC proof, and also, better handle the corner case of NTP time overflowing between the 2 timestamps passed to it.
* Address miscellaneous MISRA violations in the library.
* Hygiene change of breaking down large test functions into smaller scopes.

### CODE REVIEW

File: source/core_sntp_client.c [hunk 1]
```diff
@@ -173,6 +173,8 @@ static uint32_t calculateElapsedTimeMs( const SntpTimestamp_t * pCurrentTime,
  * - #SntpSuccess if the context is verified to be initialized.
  * - #SntpErrorBadParameter if the context is NULL.
  * - #SntpErrorContextNotInitialized if the context is validated to be initialized.
+ * - #SntpErrorChangeServer if no more servers are remaining from the configured list to
+ * synchronizing time from.
```
Comment (muneebahmed10) [inline source/core_sntp_client.c:177]: The second line should probably be justified with the rest of the bullet point. Also, "synchronizing" -> "synchronize"
```suggestion
 * - #SntpErrorChangeServer if no more servers are remaining from the configured list
 *   from which to synchronize time.
```

**Review by muneebahmed10 — APPROVED** [review]

File: source/core_sntp_client.c [hunk 2]
```diff
@@ -771,6 +762,11 @@ SntpStatus_t Sntp_ReceiveTimeResponse( SntpContext_t * pContext,
                 ( calculateElapsedTimeMs( &loopIterTime, pRequestTime ) >= pContext->responseTimeoutMs ) )
             {
                 status = SntpErrorResponseTimeout;
+
+                /* As server has timed out in sending its response, we will rotate to the next server in
+                 * the list of configured time servers. */
+                pContext->currentServerIndex++;
```
[the remark this answers is not in the document]
Comment (aggarw13) [reply source/core_sntp_client.c]: We discussed separately, and decided to have the server rotation use wrap around once it has used up all the servers so that the SNTP client functionality is not disturbed.

**Review by muneebahmed10 — APPROVED** [review]

**Review by cobusve — APPROVED** [review]
